tools/NewRelicHelper.psm1

```diff
@@ -176,101 +184,109 @@ function update_azure_service_definition([System.__ComObject] $project){
-        	foreach ($i in $modifiedConfigSettings.Setting){
-        		if ($i.name -eq "NewRelic.LicenseKey"){
-        			$nodeExists = $true
-        		}
-        	}
-        	if($NewRelicTask -eq $null -and !$nodeExists){
-        		$modifiedConfigSettings.AppendChild($settingNode)
-        	}
-        }
+            $taskNode.AppendChild($environmentNode)
+            $startupNode.AppendChild($taskNode)
         
-		$xml.Save($ServiceDefinitionConfig);
+            $modifiedStartUp = $modified.StartUp
+            if($modifiedStartUp -eq $null){
+            	$modified.PrependChild($startupNode)
+            }
+            else{
+            	$nodeExists = $false
+            	foreach ($i in $modifiedStartUp.Task){
+            		if ($i.commandLine -eq "newrelic.cmd"){
+            			$nodeExists = $true
+            		}
+            	}
+            	if($NewRelicTask -eq $null -and !$nodeExists){
+            		$modifiedStartUp.AppendChild($taskNode)
+            	}
+            }
+            
+            if($isWorkerRole -eq 'true'){
+            
+                $role = "WorkerRole"    
+                
+            	#Generate the environment variables for worker role instrumentation
+            	$runtimeNode = $xml.CreateElement('Runtime','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
+            	$runtimeEnvironmentNode = $xml.CreateElement('Environment','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
+            	
+            	#Enables profiling for all CLR based applications
+            	$variableCEPNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
+            	$variableCEPNode.SetAttribute('name','COR_ENABLE_PROFILING')
+            	$variableCEPNode.SetAttribute('value','1')
+            	
+            	#Profiler guid associated with the New Relic .NET profiler
+            	$variableCPNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
+            	$variableCPNode.SetAttribute('name','COR_PROFILER')
+            	$variableCPNode.SetAttribute('value','{71DA0A04-7777-4EC6-9643-7D28B46A8A41}')
+            	
+            	#Helps Azure Workers find the newrelic.config
+            	$variableNHNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
+            	$variableNHNode.SetAttribute('name','NEWRELIC_HOME')
+                $variableNHNode.SetAttribute('value','D:\ProgramData\New Relic\.NET Agent\')
+                
+                #Helps Azure Workers find the NewRelic.Agent.Core.dll
+            	$variableNIPNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
+            	$variableNIPNode.SetAttribute('name','NEWRELIC_INSTALL_PATH')
+                $variableNIPNode.SetAttribute('value','D:\Program Files\New Relic\.NET Agent\')
```

Comment:
Lots of string constants in this area of the code and some are repeated. Consider DRY and define as constants for the script.